drivers/include/periph/sleepmem.h

```diff
@@ -0,0 +1,12 @@
+#ifndef SLEEPMEM_H
+#define SLEEPMEM_H
+
+#include <unistd.h>
+
+extern size_t sleepmem_size;
+
+void sleepmem_init(void);
+size_t sleepmem_put(void *src, size_t n);
+size_t sleepmem_get(void *target, size_t n);
```

Comment:
The API is a bit strange though.

 - add an offset so we don't always have to write the whole memory at once
 - just return an error if the buffer won't fit, partial writes are not really useful

Reply:
>     * add an offset so we don't always have to write the whole memory at once

I didn't add an offset because I assumed that it might require extra logic if the sleep memory has to be written as a whole.



>     * just return an error if the buffer won't fit, partial writes are not really useful

totally, will fix

Reply:
> I assumed that it might require extra logic if the sleep memory has to be written as a whole.

Is that the case on STM32?
On sam0 you can just write individual registers.

Reply:
no, stm32 can also write individual registers.